### PR TITLE
Add org.freedesktop.Platform.GStreamer.openmpt

### PR DESCRIPTION
**Pull request by detjensrobert**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

---

This extension adds support for tracker-style music files to GStreamer using the `openmpt` plugin from `gst-plugins-bad`.

Some questions:
- ~~I'd like to support both current o.fd.P runtimes, but the build process for GStreamer has changed between 21.08 and 22.08.  What is the recommended way to handle building extensions against multiple branches of a runtime with different manifests?~~
  - ~~I've included a second manifest here (`...openmpt.21.08.yaml`), but a) its a second manifest and b) Meson needs network access during the build to fetch subprojects. I did not see the "correct" way to handle this in existing manifests, so I assume this will need some tweaks.~~
- Is metainfo necessary for extensions?

### CODE REVIEW

File: org.freedesktop.Platform.GStreamer.openmpt.21.08.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      - -Dauto_features=disabled
+      - -Dgood=disabled
+      - -Dugly=disabled
+      - -Dbad=enabled
+      - -Dgst-plugins-bad:openmpt=enabled
+      - -Dgst-full-plugins=openmpt
+      - --buildtype=release
+    build-options:
+      build-args:
+        - --share=network # for meson subprojects :/
```
Comment (hfiguiere) [inline org.freedesktop.Platform.GStreamer.openmpt.21.08.yaml:55]: that won't work. Build must be done offline.

**Comment by hfiguiere** [comment]
there is no point in supporting 21.08. It will be EOL in august when 23.08 is released.

**Comment by detjensrobert** [comment]
Ah, makes sense.

**Comment by detjensrobert** [comment]
bot, build org.freedesktop.Platform.GStreamer.openmpt

File: org.freedesktop.Platform.GStreamer.openmpt.metainfo.xml
Comment (hfiguiere) [inline org.freedesktop.Platform.GStreamer.openmpt.metainfo.xml:1]: this file is never installed.

**Comment by detjensrobert** [comment]
bot, build org.freedesktop.Platform.GStreamer.openmpt

**Comment by detjensrobert** [comment]
bot, build org.freedesktop.Platform.GStreamer.openmpt

**Comment by detjensrobert** [comment]
Hmm.. the manifest is creating the appstream info at `/usr/lib/extensions/gstreamer-1.0/openmpt/share/app-info/...`, but the buildbot is looking for `/*/share/app-info/`. Is the buildbot setup to look into a deeply nested `$FLATPAK_DEST`, or is the glob here only expanding to one directory?

**Comment by detjensrobert** [comment]
bot, build org.freedesktop.Platform.GStreamer.openmpt

**Comment by detjensrobert** [comment]
Test flatpak works.

<details>
<summary>Playing a <code>.s3m</code> in G4Music</summary>

image

</details>

**Pull request closed without merging**